src/slackapp/flows/editPost.js

```diff
@@ -0,0 +1,217 @@
+const {
+  errorView,
+  successView,
+  successResponse,
+  errorResponse
+} = require("../views");
+const slackapi = require("../../slackapi");
+const guard = require("../lib/guard");
+const { findChannelByName, listMembers } = require("../lib/channels");
+
+editPost.id = "edit_post";
+saveEdits.id = "save_post_edit";
+deletePost.id = "delete_post";
+
+// Members of these channels can edit messages
+const editorsAllowedChannels = ["#admin", "#intake_volunteers"];
+
+/**
+ * Adds a message shortcut to allow users to edit bot messages.
+ */
+module.exports.register = function register(slackInteractions) {
+  // Prompt user with editing dialog
+  slackInteractions.action(
+    {
+      type: "message_action",
+      callbackId: editPost.id
+    },
+    guard(editPost)
+  );
+
+  // Handle button that deletes post
+  slackInteractions.action(
+    {
+      actionId: deletePost.id
+    },
+    guard(deletePost)
+  );
+
+  // Handles actually updating the post
+  slackInteractions.viewSubmission(saveEdits.id, guard(saveEdits));
+};
+
+async function editPost(payload) {
+  const { message, channel, user } = payload;
+  if (!message) {
+    return openError(
+      payload.trigger_id,
+      "This shortcut can only be used on messages."
+    );
+  }
+
+  const botUser = await slackapi.auth.test();
+  if (message.bot_id !== botUser.bot_id) {
+    return openError(
+      payload.trigger_id,
+      "You can only edit messages posted by the bot."
+    );
+  }
+
+  const canEdit = await userCanEditMessage(user, message);
+  if (!canEdit) {
+    return openError(
+      payload.trigger_id,
+      "You aren't permitted to edit this message."
+    );
+  }
+
+  const view = await makeEditPostView(payload, message, channel);
+  return slackapi.views.open({
+    trigger_id: payload.trigger_id,
+    view
+  });
+}
+
+async function saveEdits(payload) {
+  const meta = JSON.parse(payload.view.private_metadata);
+  const { values } = payload.view.state;
+  const content = values.message.message.value;
+  try {
+    await slackapi.chat.update({
+      channel: meta.message_channel,
+      ts: meta.message_ts,
+      text: content
+    });
+  } catch (e) {
+    console.error("Error updating message %O %O", meta, e);
+    return errorResponse(`Failed to update message: ${e}`);
+  }
+  return successResponse("The message was successfully updated");
+}
+
+async function deletePost(payload) {
+  const meta = JSON.parse(payload.view.private_metadata);
+  try {
+    await slackapi.chat.delete({
+      channel: meta.message_channel,
+      ts: meta.message_ts
+    });
```

Comment:
It'd be cool but probably out of scope to clear the slack message data for a request so the bot could post again. would you mind filing a PR for this?

Reply:
Airtable formulas support string operations so we could add findInMeta to the api or something. I'll leave it up to you if you want to include in a separate PR